### PR TITLE
Add file that summarizes the FSRS scheduler

### PR DESCRIPTION
**Pull request by joshdavham**
I've been finding myself repeatedly referring to both [ishiko's ts-fsrs workflow diagram](https://github.com/open-spaced-repetition/ts-fsrs/blob/main/ts-fsrs-workflow.drawio) and the [The Algorithm page](https://github.com/open-spaced-repetition/fsrs4anki/wiki/The-Algorithm) and thought it'd be useful to include a file that gives a brief summary of how the FSRS scheduler works, at a high level.

I did omit some of the details of the implementation however, such as how a New card, when rated Again/Hard/Good, will be due in 1, 5 or 10 minutes, but I did this to keep things simple.

Please let me know if this looks okay, or if there are any issues 👍

Also, here's the [link](https://www.figma.com/board/Ak8TFr4rmglYs0CSO5xyFT/FSRS-5?node-id=0-1&t=on6qRaqIXZaT5mPO-1) to the diagram if you're interested.

### CODE REVIEW

**Comment by brainwo** [comment]
I remade the diagram using Mermaid (embeddable in GitHub Markdown):
```mermaid
---
title: FSRS State Diagram
---
graph LR;
    new([New])--->|Again|learning([Learning]);
    new([New])--->|Hard|learning([Learning]);
    new([New])--->|Good|learning([Learning]);
    new-->|Easy|review([Review]);
    learning-->|Easy|review;
    learning-->|Good|review;
    learning-->|Again|learning;
    learning-->|Hard|learning;
    relearning([Relearning])-->|Good|review;
    relearning([Relearning])-->|Easy|review;
    relearning-->|Again|relearning;
    relearning--->|Hard|relearning;
    review-->|Hard|review;
    review-->|Good|review;
    review-->|Easy|review;
    review-.->|Again*|relearning;
    style new fill:#E3CBFE,stroke:#d577ff,color:#000
    style learning fill:#bde3ff,stroke:#22b4ff,color:#000
    style relearning fill:#bde3ff,stroke:#22b4ff,color:#000
    style review fill:#aff4c6,stroke:#65bf8c,color:#000
    linkStyle 1,7,11,12 color:orange;
    linkStyle 0,6,10 color:red;
    linkStyle 15 stroke:red,color:red;
    linkStyle 3,4,9,14 color:lime;
    linkStyle 2,5,8,13 color:cyan;
```

**Comment by ishiko732** [comment]
It is important to note that this is a diagram for short-term scheduling and does not include scenarios for long-term scheduling.
During the long-term scheduling process, the (Re)Learning state does not exist.

File: ALGORITHM.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+$R(t,S) = \left(1+FACTOR\cdot \frac{t}{S}\right)^{DECAY}$
+
+<ins>Forget stability</ins>: 
+
+$S^\prime_f(D,S,R) = w_{11} \cdot D^{-12}\cdot \Big[(S+1)^{w_{13}} -1 \Big]\cdot e^{w_{14}\cdot (1-R)}$
+
+<ins>Recall stability</ins>: 
+
+$$S^\prime_r(D,S,R,G) = S\cdot \left[1+e^{w_{8}}\cdot (11-D)\cdot S^{-w_9}\cdot (e^{w_{10}\cdot (1-R)}-1)\cdot \textrm{HARD PENALTY(if $G$=2)}\cdot \textrm{EASY BONUS(if $G$=4)} \right]$$
+
```
Comment (ishiko732) [inline ALGORITHM.md:81]: The explanation for the `next_difficulty` method is missing.

$$D^\prime(D,G) =MR(D_0(4) , D - w_6 \cdot (G - 3))$$

File: ALGORITHM.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+$I = 0$
+
+**<ins>Learning / Relearning card rated Again / Hard:</ins>**
+
+State <- Learning / Relearning
+
+$S = S^\prime(S,G)$
+
+$D = MR\Big(D_0(4), D-w_6\cdot (G-3)\Big)$
```
Comment (ishiko732) [inline ALGORITHM.md:111]: The `repeat` method does not directly call `mean_reversion`. I suggest changing it to use `next_difficulty`, as follows:  
```suggestion
$D = D^\prime(D,G)$
```

File: ALGORITHM.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+$I = I(r,S)$
+
+**<ins>Review card rated Again:</ins>**
+
+State <- Relearning
+
+$S = S^\prime_f\big(D,S,R(t,S)\big)$
+
+$D = MR\Big(D_0(4), D-w_6\cdot (1-3)\Big)$
```
Comment (ishiko732) [inline ALGORITHM.md:131]:
```suggestion
$D = D^\prime(D,G)$
```

File: ALGORITHM.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+$I=0$
+
+**<ins>Review card rated Hard / Good / Easy:</ins>**
+
+State <- Review
+
+$S = S^\prime_r\big(D,S,R(t,S),G\big)$
+
+$D = MR\Big(D_0(4), D-w_6\cdot (G-3)\Big)$
```
Comment (ishiko732) [inline ALGORITHM.md:141]:
```suggestion
$D = D^\prime(D,G)$
```

**Review by ishiko732 — CHANGES_REQUESTED** [review]
Please consider whether to add the description $D \in [1,10]$.

**Comment by ishiko732** [comment]
I think it would be better to describe it using a piecewise function.

$$
S^\prime(D,S,R,G) = 
\begin{cases} 
S^\prime_f\big(D,S,R(t,S)\big) & \text{if } G = Again \\
S^\prime_r\big(D,S,R(t,S),G\big) & \text{if } G = \text{Hard,Good,Easy}
\end{cases}
$$

**Comment by joshdavham** [comment]
Thanks for the feedback. I’m currently away from my computer but I’ll be back tomorrow and can review these changes then

**Review by ishiko732 — APPROVED** [review]
LGTM 
@L-M-Sherlock  cc

**Review by L-M-Sherlock — APPROVED** [review]
That's great!

**Pull request merged**